python/cudf/cudf/core/column/string.py

```diff
@@ -3946,6 +3948,32 @@ def base_size(self):
                 / self.base_children[0].dtype.itemsize
             )
 
+    def min(self, dtype=None):
+        return libcudf.reduce.reduce("min", self, dtype=dtype)
+
+    def max(self, dtype=None):
+        return libcudf.reduce.reduce("max", self, dtype=dtype)
+
+    def sum(self, dtype=None):
+        return self.str().cat()
+
+    def product(self, dtype=None):
+        raise TypeError("can't multiply sequence by non-int of type 'object'")
+
+    def mean(self, dtype=np.float64):
+        cat_result = self.str().cat()
+        try:
+            cat_result = float(cat_result)
+        except ValueError:
+            raise TypeError(f"Could not convert {cat_result} to numeric")
+        return cat_result / len(self)
```

Comment:
Is this actually a thing we need to handle? Does Pandas allow for getting the mean of a column of strings of numeric values?

Reply:
-1 from me as this sounds like it can lead to some difficult to track bugs. Perhaps let's wait to see if there's a valid use case?

Reply:
So in case of pandas, the behavior is actually **appending** all strings of series and then trying to convert to float and dividing by the length of series.

```python
In [12]: sr = pd.Series(['1', '3', '5'])                                                                 

In [13]: sr.mean()                                                                                       
Out[13]: 45.0
```

So it is `float('135')/3`. Whenever the result of `.str.cat` cannot be type-casted to a float scalar, pandas will throw an error.

Reply:
Gotcha -- I think it's more useful to throw here. In the event that the numeric values were inadvertently cast to strings, calling `mean()` will give a believable - but clearly wrong - result.

Personally I can't think of a great use case for `'abc'.mean()`, but we can consider implementing it if we hear from users about one :)

Reply:
+1 to throwing with a meaningful error message for now and re-evaluating if there's user feedback.

Reply:
The fact that `pd.Series(['a', 'b', 'c']).mean()` throws, but `pd.Series(['1', '3', '5']).mean()` doesn't and returns `45.0` as the result feels like a bug. Could you raise a Pandas issue?

Reply:
Will do

Reply:
https://github.com/pandas-dev/pandas/issues/34957

Reply:
Okay this was confirmed a bug on the Pandas side so we should definitely throw.